### PR TITLE
Add post editor and clothing scraper

### PR DESCRIPTION
**Pull request by danielheyman**
This is what it looks like when you type a post:
screen shot 2017-08-05 at 4 23 11 pm

### CODE REVIEW

File: tests/data/clothing.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const Clothing = require('../../data/').clothing;
+const expect = require('chai').expect
+
+
+describe('Clothing data module', () => {  
+    it('should retrieve hm data', function * () {
+        const res = yield Clothing.retrieveClothingInfo("http://www.hm.com/us/product/72665");
+        expect(res.price).to.be.eql(119);
+        expect(res.image).to.be.include('http://lp.hm.com/hmprod');
+    })
```
Comment (danielheyman) [inline tests/data/clothing.js:10]: Also added the module to retrieve the price and image

Comment (danielheyman) [reply tests/data/clothing.js:10]: Works with every website I tested so far :)

**Comment by tsangjustin** [comment]
How do I play with this feature when creating a forum?

**Comment by danielheyman** [comment]
@tsangjustin I think Weronika is working on the forum creation feature. Out of the scope of this PR

**Comment by danielheyman** [comment]
I put a demo on the forum creation page if you just check that out

**Comment by tsangjustin** [comment]
Yeah, I tried the demo, but when I hit enter it justs shows me a 404 on POST /post

File: data/clothing.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ * @return  true if the string is valid; otherwise, return false
+ */
+function isValidString(str) {
+	if ((str === undefined) || (typeof(str) !== "string") || (str.length <= 0)) {
+		return false;
+	}
+	return true;
+}
+
+let user = exports = module.exports = {
```
Comment (tsangjustin) [inline data/clothing.js:17]: please rename user to clothing

Comment (danielheyman) [reply data/clothing.js:17]: Woops my bad

**Pull request merged**